### ISSUE TITLE
No selection of help topics based up on department selection

### ISSUE THREAD

**Issue by jeffchulg** [osTicket/osTicket-1.7#217]
When creating a new ticket, you select the department and help topic. There is no selection of help topics based up on department selection.  

Suppose you have a big company with a lot of help topics and relevant names for departments. This should be useful for customers, or even for staff to first select the department and get only help topics corresponding to that department (with a "general" as an option)

**Comment by protich** [comment]
At the moment osTicket simply maps help topics to departments in the backend. Departments are **not** exposed to users (client interface) at all. Staff can overwrite help topic routing by selecting a department directly - actually department selection **should** be optional.

That said, I see your point in cases of large number of help topics. The solution is not to expose departments (some are private/internal) but allow help topics nesting.

**Comment by jeffchulg** [comment]
Yes.. Help topic nesting is a good idea for this kind of usage... Maybe enhancement for further release ? :)

**Comment by jeffchulg** [comment]
I though about one more thing about help topics recently... What about a "multi-level help topic access control" : customer level and internal levels (corporate, department, ... ). 

As you ticketing system can be used either for internal or for external purpose, there could be some help topics visible from staff interface but not from client interface as it's the case now, but also help topics that are specific to a department, and only a member of that department could create or edit a ticket with this help topic. 

What is your opinon ?
